tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -219,47 +219,51 @@ static void write_kernel_data(void)
 /*
  * volatile to avoid compiler mischief.
  */
-volatile int *ptr = NULL;
+volatile int *priv_stack_ptr;
 #if defined(CONFIG_X86)
 volatile size_t size = MMU_PAGE_SIZE;
```

Comment:
Kind of baffled on why this is volatile.
Since this is now only used for x86, maybe just delete it and use MMU_PAGE_SIZE in its place.

Reply:
Pushed an updated commit that drops volatile on size and adds a comment explaining.